File: CSharp/Tests/CountSyllablesTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class CountSyllablesTest
    {
        [Theory]
        [InlineData("buf-fet", 2)]
        [InlineData("beau-ti-ful", 3)]
        [InlineData("mon-u-men-tal", 4)]
        [InlineData("on-o-mat-o-poe-ia", 6)]
        [InlineData("o-ver-ly", 3)]
        [InlineData("pas-try", 2)]
        [InlineData("flu-id", 2)]
        [InlineData("syl-la-ble", 3)]
        public void Count_StringValue_ReturnSyllableCount(string word, int expected)
        {
            var actual = CountSyllables.Count(word);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/countSyllables.test.js
const count = require("../countSyllables");

const data = [
  ["buf-fet", 2],
  ["beau-ti-ful", 3],
  ["mon-u-men-tal", 4],
  ["on-o-mat-o-poe-ia", 6],
  ["o-ver-ly", 3],
  ["pas-try", 2],
  ["flu-id", 2],
  ["syl-la-ble", 3]
];

describe("count", () => {
  describe("Return number of syllables", () => {
    test.each(data)("count(%p) = %i", (word, result) => {
      let actual = count(word);

      expect(actual).toEqual(result);
    });
  });
});
